modules/search/instant-search/components/search-app.jsx

```diff
@@ -0,0 +1,246 @@
+/** @jsx h */
+
+/**
+ * External dependencies
+ */
+import Preact, { h, Component } from 'preact';
+import Portal from 'preact-portal';
+// NOTE: We only import the debounce package here for to reduced bundle size.
+//       Do not import the entire lodash library!
+// eslint-disable-next-line lodash/import-scope
+import debounce from 'lodash/debounce';
+
+/**
+ * Internal dependencies
+ */
+import SearchResults from './search-results';
+import SearchFiltersWidget from './search-filters-widget';
+import SearchSortWidget from './search-sort-widget';
+import SearchBox from './search-box';
+import { search, buildFilterAggregations } from '../lib/api';
+import {
+	setSearchQuery,
+	setFilterQuery,
+	getFilterQuery,
+	setSortQuery,
+	getSortQuery,
+	hasFilter,
+} from '../lib/query-string';
+import { removeChildren, hideElements, hideChildren, showChildren } from '../lib/dom';
+
+class SearchApp extends Component {
+	constructor() {
+		super( ...arguments );
+		this.input = Preact.createRef();
+		this.requestId = 0;
+
+		// TODO: Rework these lines. We shouldn't reassign properties.
+		this.props.resultFormat = 'minimal';
+		this.props.aggregations = buildFilterAggregations( this.props.options.widgets );
+		this.props.widgets = this.props.options.widgets ? this.props.options.widgets : [];
+		this.isSearchPage = this.props.initialValue !== '';
+
+		this.state = {
+			isLoading: false,
+			query: this.props.initialValue,
+			response: {},
+			sort: this.props.initialSort,
+		};
+		this.getDebouncedResults = debounce( this.getResults, 200 );
+
+		//clean up the page in prep for adding component
+		// we can manipulate the existing DOM on the page, but not this component because
+		// it hasn't mounted yet.
+		hideElements( this.props.themeOptions.elem_selectors );
+		if ( this.hasActiveQuery() ) {
+			this.activateResults();
+		}
+		this.props.widgets.forEach( function( widget ) {
+			removeChildren( document.getElementById( widget.widget_id ) );
```

Comment:
This doesn't quite work for me:
https://jeremy.hu/?s=marvel

```
Uncaught TypeError: Cannot read property 'lastChild' of null
    at removeChildren (dom.js:2)
    at search-app.jsx:59
```

I do have a widget in the footer though.